python/ray/_private/import_thread.py

```diff
@@ -75,22 +70,27 @@ def _run(self):
                 if msg is None:
                     self.threads_stopped.wait(timeout=0.01)
                     continue
-
                 if msg["type"] == "subscribe":
                     continue
-                assert msg["data"] == b"rpush"
-                num_imports = self.redis_client.llen("Exports")
-                assert num_imports >= num_imported
-                for i in range(num_imported, num_imports):
-                    num_imported += 1
-                    key = self.redis_client.lindex("Exports", i)
-                    self._process_key(key)
+                self._do_importing()
         except (OSError, redis.exceptions.ConnectionError, grpc.RpcError) as e:
             logger.error(f"ImportThread: {e}")
         finally:
             # Close the pubsub client to avoid leaking file descriptors.
             import_pubsub_client.close()
 
+    def _do_importing(self):
+        while True:
+            export_key = ray._private.function_manager.make_export_key(
+                self.num_imported + 1)
+            key = self.gcs_client.internal_kv_get(
```

Comment:
Isn't this a KV op for every key, even those we don't need in this job?

Reply:
Yes, but it's the same as before:

```
                assert msg["data"] == b"rpush"
                num_imports = self.redis_client.llen("Exports")
                assert num_imports >= num_imported
                for i in range(num_imported, num_imports):
                    num_imported += 1
                    key = self.redis_client.lindex("Exports", i)
                    self._process_key(key)
```